### PR TITLE
Improve goBack function

### PR DESCRIPTION
**Pull request by adamgrzybowski**
### Details
This PR improves `goBack` function so the fallback could be used for reports screens. Additionally, it allows to `goBack` with more than one route for report screens when using a fallback route
### Fixed Issues

$ https://github.com/Expensify/App/issues/26569
PROPOSAL: 

### Tests

1. Open a chat with somebody
2. Request money
3. Open the request money report
4. Delete request
5. Go back
6. Make sure that you are not on the "not found" page

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/67908363/996d45ee-2e24-454b-89a0-7c684d68f6c2

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/67908363/d5d084ef-9a2e-4e64-9793-16b3fefa123c

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/67908363/8bc2fac1-eea6-4bda-bbc9-66d68ce238ed

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/67908363/2a5d49a8-a01c-4b3a-824f-efc16583ff35

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/67908363/fb5bc0e9-d326-449b-913d-b9085ad12534

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/67908363/65a73785-90d6-4103-bcf0-7d64d8fb45f6

</details>

### CODE REVIEW

**Comment by mountiny** [comment]
@sobitneupane putting it on hold but please continue with a review and testing please thanks!

**Comment by ShogunFire** [comment]
Hello, are you close to merging this ?

**Comment by mountiny** [comment]
@adamgrzybowski can you please resolve conflicts on this one?

**Comment by mountiny** [comment]
@adamgrzybowski bump on the conflicts

**Comment by adamgrzybowski** [comment]
@mountiny done! Thanks for the bump 😄 I completely forgot

**Comment by mountiny** [comment]
@adamgrzybowski jest units failing now, maybe we need to merge main again 🤔

**Comment by adamgrzybowski** [comment]
@mountiny merging main and small adjustments in tests helped. Thanks for noticing the problem! 🙇

**Comment by mountiny** [comment]
@sobitneupane can you please do the checklist today?

**Comment by sobitneupane** [comment]
Yes @mountiny. I will review the PR today.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/25876548/5e882666-29ba-4cda-92ee-58524a7f7084


</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://github.com/Expensify/App/assets/25876548/99c5201f-e713-4493-be17-421d4cf5ac31



</details>

<details>
<summary>Mobile Web - Safari</summary>



https://github.com/Expensify/App/assets/25876548/d0d79543-9a54-45c9-9ce5-00ca9f09662b

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/118aefb5-688b-4d43-a0cb-0adde438bb15

</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/25876548/54c81d5a-ef90-437c-81a5-2e4b985ecca6


</details>

<details>
<summary>Android</summary>



https://github.com/Expensify/App/assets/25876548/aff07cc6-dd69-4621-ba90-7f95a334bbfc

</details>

**Review by sobitneupane — COMMENTED** [review]
@adamgrzybowski Is it expected that we can navigate to the deleted report by clicking 'go forward' in the browser?

If the transaction report is accessed directly through a deep link, the issue can still be reproduced.

**Comment by mountiny** [comment]
> If the transaction report is accessed directly through a deep link, the issue can still be reproduced.

What do you mean exactly, can you write out the steps?

> Is it expected that we can navigate to the deleted report by clicking 'go forward' in the browser?

I think forward considerations are out of scope

**Comment by mountiny** [comment]
@sobitneupane Are you able to finish up the review?

**Comment by s77rt** [comment]
> If the transaction report is accessed directly through a deep link, the issue can still be reproduced.

I think this may be okay as on deep link we have no other routes besides the deep link. [State is not persisted between refreshes/app reopen.](https://reactnavigation.org/docs/state-persistence/)

https://github.com/Expensify/App/blob/28b30fc102bd41ee6961785bc2dbc22aae52e422/src/libs/Navigation/Navigation.js#L134-L137

**Comment by mountiny** [comment]
@s77rt can you do the review and checklist please? Thanks!

File: src/libs/Navigation/Navigation.js
Comment (s77rt) [inline src/libs/Navigation/Navigation.js:89]: I think this may cause a performance degradation for users who keep using the app for long periods. The more you use the app the more paths we have to extract. Most of the paths we get will be a waste. I to implement an early return mechanism.

```js
function getIndexOfPathInRootNavigator(path) {
    let currentState = navigationRef.getRootState();

    for (let index = 0; index < 5; index++) {
        if (!currentState.routes.length) {
            break;
        }

        const pathFromState = getPathFromState(currentState, linkingConfig.config);
        if (path === pathFromState.substring(1)) {
            return index;
        }

        currentState = {...currentState, routes: currentState.routes.slice(0, -1), index: currentState.index - 1};
    }

    return -1;
} 
```



The 5 limit is used as a threshold. There would be no case where we would pop more than 4 screens. Currently I think the max to pop is 2. So maybe drop the 5 to 3.

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by adamgrzybowski** [comment]
Hey @s77rt, sorry for the late response to your comments. I was absorbed by the global nav 😄 

You made a very good point about the performance of this solution. I used your suggestion to make adjustments to this PR.  Thanks a lot! 🙇 

@mountiny for your attention and pulling this PR out of limbo

**Comment by s77rt** [comment]
@adamgrzybowski Can you please check the following bug (not reproducible on main)


https://github.com/Expensify/App/assets/16493223/f8ff5bc6-56fd-447d-b601-fd7e85e276e9

**Comment by s77rt** [comment]
Please let me know when this is ready review/retest

**Comment by WojtekBoman** [comment]
@s77rt It's ready

**Comment by mountiny** [comment]
@s77rt are you able to do the review here please?

**Comment by s77rt** [comment]
@mountiny Yes, checking now

File: src/pages/home/report/ReportActionsList.js
Comment (s77rt) [inline src/pages/home/report/ReportActionsList.js:163]: This does feel like a workaround. I also remember this bug being related to the use of Freeze or is this bug related only to this PR (i.e. not reproducible in main)?

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16493223/0cbf45e9-6a7c-4713-ae4d-6a01cb7818f4


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16493223/e794d9d6-5806-4272-a548-7bdeeee213d1


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16493223/d485059a-a1dc-4ade-a2b7-4c0d310ce306


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16493223/9f0c8473-d690-4da8-8b0a-33c86cf3198f


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/16493223/13b4cbe2-ebb1-4070-a649-715f539ccb5a


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16493223/54da8273-e8c2-4b64-9c54-f2b7a10f617b


</details>

**Review by s77rt — APPROVED** [review]
Overall looks good, just need to address https://github.com/Expensify/App/pull/26498#discussion_r1386932481 before merging

File: src/pages/home/report/ReportActionsList.js
Comment (WojtekBoman) [reply src/pages/home/report/ReportActionsList.js:163]: It is connected with the fact that now we go back to the screen which is already on the navigation stack. On the main branch we invoke two navigate functions after delete the request money. In the result we render flatList for the first time and this bug doesn't occur. Now instead of pushing new screen to the navigation stack we go back to the existing one. After the deletion process the flatList doesn't automatically scroll to the bottom, there is a free space left after the deleted item. This hook solves this problem

Comment (s77rt) [reply src/pages/home/report/ReportActionsList.js:163]: I removed the hook and was not able to reproduce. Can you please double check?


https://github.com/Expensify/App/assets/16493223/e7ba8a7e-85bf-4995-8067-f10eccab933c

Comment (WojtekBoman) [reply src/pages/home/report/ReportActionsList.js:163]: Could you check it on the mobile platforms? On web this bug doesn't occur

Comment (s77rt) [reply src/pages/home/report/ReportActionsList.js:163]: Same 

https://github.com/Expensify/App/assets/16493223/20f80554-8c82-4357-a34f-c0cb9ee6f843

**Comment by mountiny** [comment]
Could you please merge main into the branch to hoopefully resolve the reassure tests

File: src/pages/home/report/ReportActionsList.js
Comment (WojtekBoman) [reply src/pages/home/report/ReportActionsList.js:163]: https://github.com/Expensify/App/assets/47774969/9f2c6732-03e9-41c2-a968-bdf661a469ce

I've checked it again and I am able to reproduce it. Before the test I pulled the last changes from main

Comment (s77rt) [reply src/pages/home/report/ReportActionsList.js:163]: The bug is only reproducible on iOS


https://github.com/Expensify/App/assets/16493223/8c25aa3c-bc8f-49a9-8015-babf395917f7

Comment (s77rt) [reply src/pages/home/report/ReportActionsList.js:163]: I think the root cause is the autoscrollToTopThreshold value in maintainVisibleContentPosition

**Comment by mountiny** [comment]
@s77rt What is the latest on the bug, what was the resolution on the Pr you ave linked?

@WojtekBoman lets merge main again since the reassure tests has been fixed there

**Comment by s77rt** [comment]
The bug is due to the use of `maintainVisibleContentPosition`. We had a similar issue https://github.com/Expensify/App/issues/30423 that was fixed by https://github.com/Expensify/App/pull/30461. It seems that we need to adjust the value of `maintainVisibleContentPosition.autoScrollToTopThreshold` once again but not sure what value should we use. It may be easier to handle this bug separately and merge this PR (and remove the scroll hack)

**Comment by WojtekBoman** [comment]
I removed this hook from `ReportActionsList` file

**Comment by mountiny** [comment]
@s77rt could you please retest

**Comment by s77rt** [comment]
@mountiny Just did. Still working as expected but the bug on iOS is still present. Should we create a new issue for that bug?

**Comment by s77rt** [comment]
After checking further more I think we can adjust the value in this PR and fix the bug.


`maintainVisibleContentPosition.autoscrollToTopThreshold` sets the threshold that if the user scrolls beyond that position the auto scroll won't work. Currently it's 64px (variables.listItemHeightNormal) which means if user scrolls or if the height is changed by anything over 64px then the auto scroll won't work. In our case we delete the IOU preview which is 123px in height. 

I don't know how we ended up choosing 64px (cc @roryabraham) but it seems too low. Let's create a new variable and set it to 128px. That should do it.

**Comment by s77rt** [comment]
@WojtekBoman Thanks! This is looking good for me

**Comment by s77rt** [comment]
@mountiny Let's merge this 🚀

**Review by mountiny — APPROVED** [review]
Thanks, I am approving but i would love to see @roryabraham opinion on this before merging

**Comment by roryabraham** [comment]
> Currently [maintainVisibleContentPosition.autoScrollToTopThreshold is] 64px ... 

This was chosen as it's the normal height of a `ReportActionItemSingle`. When it was added there was a bug when `Uploading attachment...` was replaced by the image thumbnail. 64px was just enough to fix that bug, and there wasn't any bug with sending IOU requests as far as I remember.

Increasing the `autoScrollToTopThreshold` to a higher value seems fine to me. But it's not really a perfect solution. `ReportActionItem` components have non-deterministic height, so if you choose any value, I can write you a message that's taller and if you delete it the same bug will be reproducible. So maybe we could look for a better solution. You don't want to `autoScrollToTop` if the reportAction in question (at the bottom) is not even visible on the screen. So maybe a reasonable value for `autoScrollToTopThreshold` is the height of the screen?

**Review by roryabraham — COMMENTED** [review]
> Currently [maintainVisibleContentPosition.autoScrollToTopThreshold is] 64px ... 

This was chosen as it's the normal height of a `ReportActionItemSingle`. When it was added there was a bug when `Uploading attachment...` was replaced by the image thumbnail. 64px was just enough to fix that bug, and there wasn't any bug with sending IOU requests as far as I remember.

Increasing the `autoScrollToTopThreshold` to a higher value seems fine to me. But it's not really a perfect solution. `ReportActionItem` components have non-deterministic height, so if you choose any value, I can write you a message that's taller and if you delete it the same bug will be reproducible. So maybe we could look for a better solution. You don't want to `autoScrollToTop` if the reportAction in question (at the bottom) is not even visible on the screen. So maybe a reasonable value for `autoScrollToTopThreshold` is the height of the screen?

**Comment by mountiny** [comment]
Thank you!

> So maybe a reasonable value for autoScrollToTopThreshold is the height of the screen?

Given the explanation I think this makes sense, this would have to be dynamic then. @s77rt @WojtekBoman Lets try this out and test if there are any regressions from it. I hink that can be handled in a separate PR though, do you guys agree?

**Comment by s77rt** [comment]
Handling this in a separate PR sounds good

**Comment by mountiny** [comment]
@s77rt ddi you retest after the latest change?

**Comment by s77rt** [comment]
@mountiny Yes (the 128px change - not the height-screen change)

**Review by mountiny — APPROVED** [review]
Thanks guys! Lets be on a look out for any regressions in the go back logic

File: src/components/InvertedFlatList/BaseInvertedFlatList.js
Comment (shubham1206agra) [inline src/components/InvertedFlatList/BaseInvertedFlatList.js:8]: This PR forgot to cover an extra case. More details here https://github.com/Expensify/App/issues/30726

**Pull request merged**
